AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage.py

```diff
@@ -45,144 +72,77 @@ def ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage():
 
-    def validate_start_state(entity: EditorEntity, expected_state: str):
+    def setup_level_entities(self):
+
+        self.create_editor_entity(activated_dict)
+        self.create_editor_entity(deactivated_dict)
+        self.create_editor_entity(controller_dict)
+
+    def create_editor_entity(self, entity_dict: dict):
         """
-        Validate that the starting state of the entity is correct, if it isn't then attempt to rectify and recheck.
-        :return: bool: Whether state is set as expected
+        Create entities and configure their activation status and variables. We call two different libraries to modify the
+        entity. Hydra editor utils to create the entity and editor entity utils to drill into its properties to change values.
+
         """
-        state_options = {
-            "active": azlmbr.globals.property.EditorEntityStartStatus_StartActive,
-            "inactive": azlmbr.globals.property.EditorEntityStartStatus_StartInactive,
-            "editor": azlmbr.globals.property.EditorEntityStartStatus_EditorOnly,
-        }
-        if expected_state.lower() not in state_options.keys():
-            raise ValueError(f"{expected_state} is an invalid option; valid options: active, inactive, or editor.")
-
-        state = entity.get_start_status()
-        if state != state_options[expected_state]:
-            # If state fails to set, set_start_status will assert
-            entity.set_start_status(expected_state)
-        return True
-
-    def validate_entities_in_level():
-        controller = validate_entity_exist(controller_dict["name"], Tests.controller_exists)
-        state1_correct = validate_start_state(controller, controller_dict["status"])
-
-        act_tester = validate_entity_exist(activated_dict["name"], Tests.activated_exists)
-        state2_correct = validate_start_state(act_tester, activated_dict["status"])
-
-        deac_tester = validate_entity_exist(deactivated_dict["name"], Tests.deactivated_exists)
-        state3_correct = validate_start_state(deac_tester, deactivated_dict["status"])
+        entity = scripting_tools.create_entity_with_sc_component_asset(entity_dict["name"], entity_dict["path"])
+        scripting_tools.change_entity_start_status(entity_dict["name"], entity_dict["status"])
+
+        if entity_dict["name"] == "Controller":
+            scripting_tools.change_entity_sc_variable_entity(entity_dict["name"], activated_dict["name"], ENTITY_TO_ACTIVATE_PATH)
+            scripting_tools.change_entity_sc_variable_entity(entity_dict["name"], deactivated_dict["name"], ENTITY_TO_DEACTIVATE_PATH)
+
+        helper.wait_for_condition(lambda: entity is not None, WAIT_TIME_3)
+
+    def validate_entities_in_level(self):
+
+        controller = scripting_tools.validate_entity_exists_by_name(controller_dict["name"], Tests.controller_exists)
+        state1_correct = scripting_tools.validate_entity_start_state_by_name(controller, controller_dict["status"])
```

Comment:
This code repeats for each entity you're trying to validate.

I'd recommend extracting this behavior into another function then just pass in the values you're trying to validate.

This will reduce the number of lines of code, remove repeated code, and make it clearer what you're trying to do.

Reply:
broke this down into two new functions and made the function calling the tools file a bit more generic